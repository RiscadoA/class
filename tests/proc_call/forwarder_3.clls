proc overengineered_forward(x:wait, y:close){
    cut {
        println("send");
        send ch(x);
        println("lhs fwd");
        fwd ch y
        |ch: recv wait; close|
        println("recv");
        recv ch(a);
        println("rhs fwd");
        fwd a ch
    }
};;

proc main() {
    cut {
        println("close");
        close x
        |x:wait|
        overengineered_forward(x, y)
        |y:wait|
        println("wait");
        wait y; ()
    }
};;
