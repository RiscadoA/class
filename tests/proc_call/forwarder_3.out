close
send
recv
rhs fwd
lhs fwd
wait
